examples/scripts/learning_regularization_parameter_maps_for_tv_reconstruction.py

```diff
@@ -0,0 +1,881 @@
+
+n_coils = 8
+acceleration_factor = 4.0
+noise_variance = 0.05
+
+kdata_, forward_operator_, adjoint_recon_, pseudo_inverse_solution_, image_ = prepare_data(
+    image_, n_coils=n_coils, acceleration_factor=acceleration_factor, noise_variance=noise_variance
+)
+if n_coils > 1:
+    assert pseudo_inverse_solution_ is not None
+    images_list = [adjoint_recon_.squeeze(), pseudo_inverse_solution_.squeeze(), image_.squeeze()]
+    titles = ['Adjoint', 'Pseudo-Inverse', 'Target']
+else:
+    images_list = [adjoint_recon_.squeeze(), image_.squeeze()]
+    titles = ['Adjoint', 'Target']
+show_images(
+    *images_list,
+    titles=titles,
+    show_mse=True,
+    clim=(0, 0.8 * image_.abs().max().item()),
+)
+
+# %% [markdown]
+# ## Network training
+
+# %% [markdown]
+# We now define the unrolled PDHG network by instantiating the `AdaptiveTVNetwork2D` class.
+
+# %%
+torch.manual_seed(2025)
+lambda_map_network = ParameterMapNetwork2D(n_filters=32)
+n_iterations = 128
+
+recon_matrix = mrpro.data.SpatialDimension(z=1, y=images.shape[-2], x=images.shape[-1])
+encoding_matrix = mrpro.data.SpatialDimension(z=1, y=images.shape[-2], x=images.shape[-1])
+
+adaptive_tv_network = AdaptiveTVNetwork2D(
+    recon_matrix=recon_matrix,
+    encoding_matrix=encoding_matrix,
+    lambda_map_network=lambda_map_network,
+    n_iterations=n_iterations,
+)
+
+if n_coils > 1:
+    assert pseudo_inverse_solution_ is not None
+    input_image_ = pseudo_inverse_solution_
+else:
+    input_image_ = adjoint_recon_
+
+if torch.cuda.is_available():
+    adaptive_tv_network = adaptive_tv_network.cuda()
+    input_image_ = input_image_.cuda()
+    kdata_ = kdata_.cuda()
+    forward_operator_ = forward_operator_.cuda()
+
+with torch.no_grad():
+    regularization_parameter_map_init = adaptive_tv_network.estimate_lambda_map(input_image_)
+
+
+# %% [markdown]
+# Let us now train the unrolled PDHG network. We set up the optimizer with an appropriate learning rate, choose a
+# number of epochs and write a simple training loop, in which we present the network input-target pairs and update
+# the network's parameters by gradient-descent.
+# To obtain a somewhat decent estimate of the network parameters, training should take approximately 5-6 minutes
+# on a GPU. If you have GPU and you need a break, start the training now, grab yourself a coffee and come back later
+# to see your network trained from scratch.
+# If you do not have a GPU, we provide a pre-trained model, which you can load and directly apply to the data.
+
+# %%
+# cnn_block = cast(torch.nn.Module, adaptive_tv_network.lambda_map_network.cnn_block)
+# cnn_block =
+optimizer = torch.optim.Adam(
+    [
+        {'params': adaptive_tv_network.lambda_map_network.cnn_block.parameters(), 'lr': 1e-4},
+        {'params': [adaptive_tv_network.lambda_map_network.t], 'lr': 1e-2},
+    ],
+    weight_decay=1e-8,
+)
+
+lr_scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.9)
+
+import copy
```

Comment:
should this be a new cell?

can we give some text in between the code to have it in a similar style as the other notebooks please?